beacon-chain/rpc/prysm/v1alpha1/validator/assignments.go

```diff
@@ -178,6 +184,45 @@ func (vs *Server) duties(ctx context.Context, req *ethpb.DutiesRequest) (*ethpb.
 			vStatus, _ := vs.validatorStatus(ctx, s, pubKey)
 			assignment.Status = vStatus.Status
 		}
+
+		// Are the validators in current or next epoch sync committee.
+		if postAltairTransition {
+			syncCommPeriod := helpers.SyncCommitteePeriod(req.Epoch)
+			csc, err := s.CurrentSyncCommittee()
+			if err != nil {
+				return nil, status.Errorf(codes.Internal, "Could not get current sync committee: %v", err)
+			}
+			assignment.IsSyncCommittee, err = helpers.IsCurrentPeriodSyncCommittee(s, idx)
+			if err != nil {
+				return nil, status.Errorf(codes.Internal, "Could not determine current epoch sync committee: %v", err)
+			}
+			if assignment.IsSyncCommittee {
+				assignValidatorToSyncSubnet(req.Epoch, syncCommPeriod, pubKey, csc, assignment.Status)
+			}
+
+			nextSlotEpoch := helpers.SlotToEpoch(s.Slot() + 1)
```

Comment:
`GetDuties` is only called once every epoch at slot 0, so we would never be able to check this in time. Once duties are cached, we never request them again.

Reply:
We shouldn't be fixated by caller behavior though. If someone were to use this and request slot - 1 then they will catch a bug. I would prefer the correct implementation here